jaeger-core/src/main/java/com/uber/jaeger/PropagationRegistry.java

```diff
@@ -0,0 +1,65 @@
+package com.uber.jaeger;
+
+import com.uber.jaeger.propagation.Extractor;
+import com.uber.jaeger.propagation.Injector;
+import io.opentracing.propagation.Format;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+class PropagationRegistry {
+  private final Map<Format<?>, Injector<?>> injectors = new HashMap<Format<?>, Injector<?>>();
+  private final Map<Format<?>, Extractor<?>> extractors = new HashMap<Format<?>, Extractor<?>>();
+
+  @SuppressWarnings("unchecked")
+  <T> Injector<T> getInjector(Format<T> format) {
+    return new ExceptionCatchingInjectorDecorator<T>((Injector<T>) injectors.get(format));
```

Comment:
there's no reason to wrap the injector on every `get`, instead wrap it on `register`